### PR TITLE
refactor(binder): refactor BoundSelect

### PR DESCRIPTION
**Pull request by st1page**
## What's changed and what's your intention?
- add schma method for `BoundQuery` `BoundSetExpr` `BoundSelect ` `BoundValues`
- remove the `names` `data_types` and other method for the above struct
- add construct function for `BoundSelect`, and ban constructing it with values directly, for its `schema` and `select_items` should be consistent
## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

#1958

### CODE REVIEW

File: src/frontend/src/binder/select.rs [hunk 1]
```diff
@@ -34,25 +34,43 @@ pub struct BoundSelect {
...
     pub group_by: Vec<ExprImpl>,
-    pub schema: Schema,
+    schema: Schema,
 }
 
 impl BoundSelect {
-    /// The names returned by this [`BoundSelect`].
-    pub fn names(&self) -> Vec<String> {
-        self.aliases
+    pub fn create(
```
Comment (xiangjinwu) [inline src/frontend/src/binder/select.rs:41]: Feels unnecessary to have a separate `pub create`. The only place to create `BoundXxx` is always `bind_xxx`.

File: src/frontend/src/binder/query.rs [hunk 2]
```diff
@@ -33,19 +33,14 @@ pub struct BoundQuery {
...
+    /// The schema returned by this [`BoundQuery`].
+    pub fn schema(&self) -> &Schema {
+        self.body.schema()
     }
 
     /// The types returned by this [`BoundQuery`].
     pub fn data_types(&self) -> Vec<DataType> {
-        self.body.data_types()
+        self.schema().data_types()
     }
```
Comment (xiangjinwu) [inline src/frontend/src/binder/query.rs:44]: Shall we remove this to be consistent with others?

**Review by xiangjinwu — COMMENTED** [review]
Thanks for the refactor! We definitely should not have these duplications.

File: src/frontend/src/binder/query.rs (hunk 2 above)
Comment (st1page) [reply src/frontend/src/binder/query.rs:44]: I think it might be ok, because the `BoundQuery` is the top of the Bound tree.

Comment (xiangjinwu) [reply src/frontend/src/binder/query.rs:44]: Not a big deal. But someone may still add `names` on `BoundQuery` or `data_types`+`names` on `BoundSetExpr`/... later for the same reason: `.data_types()` is shorter than `.schema().data_types()`. Anyways we still achieve the goal on making `select.schema` consistent with `select_exprs` and `aliases`.

**Review by xxchan — APPROVED** [review]
lgtm

File: src/frontend/src/binder/select.rs (hunk 1 above)
Comment (xxchan) [reply src/frontend/src/binder/select.rs:41]: +1

**Pull request merged**